src/vcpkg-test/versions.cpp

```diff
@@ -0,0 +1,22 @@
+#include <vcpkg-test/util.h>
+
+#include <vcpkg/versions.h>
+
+using namespace vcpkg;
+
+TEST_CASE ("parse version", "[versions]")
+{
+    CHECK(Version::parse("").value_or_exit(VCPKG_LINE_INFO) == Version{});
```

Comment:
I disagree; value or exit assertion failures are easy, since they print the exact line that's a problem. The only thing you get out of trying to eliminate them is when you have several test failures that you could get all at once, but our unit tests are already fast enough that it does not make sense to worry about anything beyond 'everything passed' or 'everything's doomed'

Reply:
> but our unit tests are already fast enough that it does not make sense to worry about anything beyond 'everything passed' or 'everything's doomed'

It's more complicated than that and not just about speed. If you can see all the unit test failures, you can find the "simplest" unit test that fails and work on that one first. This gives you the most information to identify the fault. However, if the first unit test that runs causes the entire harness to abort, you lose a lot of "diagnostic power".

I have personally found tests that abort the entire process to be very frustrating for this reason and took steps to first make them not abort.

> since they print the exact line that's a problem.

They print the exact line _of the assertion that fails_, but certainly not the line inside the code where the problem originates. If you just want to break on the first assertion, that's easily done via `vcpkg-test -b`[1] which also supports `CHECK()`s that _aren't_ `value_or_exit()`..

[1]:  https://catch2-temp.readthedocs.io/en/latest/command-line.html#breaking-into-the-debugger

Reply:
> have personally found tests that abort the entire process to be very frustrating for this reason and took steps to first make them not abort.

The status quo is that lots of tests see ExpectedL and that means lots of tests can't meaningfully continue. And lots of tests call value_or_exit.

There were tests that once upon a time called CHECK(foo.has_value()) but then couldn't do anything meaningful, so such a check was pointless.

>They print the exact line of the assertion that fails, but certainly not the line inside the code where the problem originates. 

The assertion that fails is better than what other parts of the test harness do to avoid value_or_exit. Hoisting this test into another function that does the value_or_default-ness is common and destroys context as to which test failed.

>that's easily done via vcpkg-test -b[1](https://catch2-temp.readthedocs.io/en/latest/command-line.html#breaking-into-the-debugger) which also supports CHECK()s that aren't value_or_exit().

That's not what PR build does so I view it as of limited value. If this is a local run, I can easily run under a debugger and stop on whatever I want. The time when the console output matters is when you can't debug it because it happens on a platform annoying to test.

I'm not opposed in general to some solution that avoids value_or_exit here but I am opposed to any that lose context as to which line fails, and as far as I know we don't have tech in Catch2 to fix that.